app/model/commands/PublishAtomCommand.scala

```diff
@@ -40,8 +40,11 @@ case class PublishAtomCommand(id: String, override val stores: DataStores, youtu
 
     getActiveAsset(previewAtom) match {
       case Some(asset) if asset.platform == Youtube =>
-        val atomWithDuration = previewAtom.copy(duration = youtube.getDuration(asset.id))
-        updateYouTube(atomWithDuration, asset).map(atomWithYoutubeUpdates => {
+        val duration = youtube.getDuration(asset.id)
+        val blockAds = if (duration.getOrElse(0L) < youtube.minDurationForAds) false else previewAtom.blockAds
```

Comment:
Indeed. `youtube.getDuration(asset.id)` returns `None` if there was a problem getting the video from the YT Data API. The chances of a call to the Partner API succeeding at this point is very low...